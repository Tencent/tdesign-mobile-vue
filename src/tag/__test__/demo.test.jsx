/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import checkableVue from '@/tag/demos/checkable.vue';
import closableVue from '@/tag/demos/closable.vue';
import mobileVue from '@/tag/demos/mobile.vue';
import sizeVue from '@/tag/demos/size.vue';
import themeVue from '@/tag/demos/theme.vue';
import typeVue from '@/tag/demos/type.vue';

const mapper = {
  checkableVue,
  closableVue,
  mobileVue,
  sizeVue,
  themeVue,
  typeVue,
};

describe('Tag', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Tag ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
