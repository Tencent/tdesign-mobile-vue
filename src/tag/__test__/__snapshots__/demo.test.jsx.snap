// Vitest Snapshot v1

exports[`Tag > Tag checkableVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <span
    class="t-tag t-tag--checkable t-tag--square t-tag--medium t-is-checked t-tag--checked"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      选中
      
    </span>
    <!--v-if-->
  </span>
  <span
    class="t-tag t-tag--checkable t-tag--square t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      未选中
      
    </span>
    <!--v-if-->
  </span>
  <span
    aria-disabled="true"
    class="t-tag t-tag--checkable t-tag--square t-tag--medium t-is-disabled t-tag--disabled"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      不可选
      
    </span>
    <!--v-if-->
  </span>
  
</div>
`;

exports[`Tag > Tag closableVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <span
    aria-disabled="false"
    class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium t-is-closable t-tag--closable"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      标签
      
    </span>
    <span
      class="t-tag__icon-close"
    >
      <svg
        class="t-icon t-icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
  </span>
  <span
    aria-disabled="false"
    class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium t-is-closable t-tag--closable"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <svg
        class="t-icon t-icon-app"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M11.25 1.75a3 3 0 100 6 3 3 0 000-6zm-2 3a2 2 0 114 0 2 2 0 01-4 0zM2 3a1 1 0 011-1h3.5a1 1 0 011 1v3.5a1 1 0 01-1 1H3a1 1 0 01-1-1V3zm1 0v3.5h3.5V3H3zM2 9.5a1 1 0 011-1h3.5a1 1 0 011 1V13a1 1 0 01-1 1H3a1 1 0 01-1-1V9.5zm1 0V13h3.5V9.5H3zM8.5 9.5a1 1 0 011-1H13a1 1 0 011 1V13a1 1 0 01-1 1H9.5a1 1 0 01-1-1V9.5zm1 3.5H13V9.5H9.5V13z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag__text"
    >
      
      标签
      
    </span>
    <span
      class="t-tag__icon-close"
    >
      <svg
        class="t-icon t-icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
  </span>
  
</div>
`;

exports[`Tag > Tag ellipsisVue demo works fine 1`] = `
<span
  aria-disabled="false"
  class="t-tag t-tag--default t-tag--square t-tag--dark t-tag--medium"
  role="button"
  style="max-width: 130px;"
>
  <span
    class="t-tag__icon"
  >
    <!---->
  </span>
  <span
    class="t-tag__text"
  >
    
    听说超长可以省略听说超长
    
  </span>
  <!--v-if-->
</span>
`;

exports[`Tag > Tag iconVue demo works fine 1`] = `
<span
  aria-disabled="false"
  class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium"
  role="button"
>
  <span
    class="t-tag__icon"
  >
    <svg
      class="t-icon t-icon-app"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M11.25 1.75a3 3 0 100 6 3 3 0 000-6zm-2 3a2 2 0 114 0 2 2 0 01-4 0zM2 3a1 1 0 011-1h3.5a1 1 0 011 1v3.5a1 1 0 01-1 1H3a1 1 0 01-1-1V3zm1 0v3.5h3.5V3H3zM2 9.5a1 1 0 011-1h3.5a1 1 0 011 1V13a1 1 0 01-1 1H3a1 1 0 01-1-1V9.5zm1 0V13h3.5V9.5H3zM8.5 9.5a1 1 0 011-1H13a1 1 0 011 1V13a1 1 0 01-1 1H9.5a1 1 0 01-1-1V9.5zm1 3.5H13V9.5H9.5V13z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </span>
  <span
    class="t-tag__text"
  >
    
    标签
    
  </span>
  <!--v-if-->
</span>
`;

exports[`Tag > Tag mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Tag 标签
  </h1>
  <p
    class="summary"
  >
    用于标记表示主体属性、类型、状态等，由底部图形和标签文字组成
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <h2
      class="tdesign-mobile-demo-block__title"
    >
      01 类型
    </h2>
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      基础标签
    </p>
    
    <div
      class="tag-demo"
    >
      
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          标签
          
        </span>
        <!--v-if-->
      </span>
      <span
        aria-disabled="false"
        class="t-tag t-tag--danger t-tag--square t-tag--dark t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          标签
          
        </span>
        <!--v-if-->
      </span>
      <span
        aria-disabled="false"
        class="t-tag t-tag--warning t-tag--square t-tag--dark t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          标签
          
        </span>
        <!--v-if-->
      </span>
      <span
        aria-disabled="false"
        class="t-tag t-tag--success t-tag--square t-tag--dark t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          标签
          
        </span>
        <!--v-if-->
      </span>
      <span
        aria-disabled="false"
        class="t-tag t-tag--default t-tag--square t-tag--dark t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          标签
          
        </span>
        <!--v-if-->
      </span>
      
    </div>
    <div
      class="tag-demo"
    >
      
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          深色
          
        </span>
        <!--v-if-->
      </span>
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--square t-tag--light t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          浅色
          
        </span>
        <!--v-if-->
      </span>
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--square t-tag--outline t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          描边
          
        </span>
        <!--v-if-->
      </span>
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--square t-tag--light-outline t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          浅色描边
          
        </span>
        <!--v-if-->
      </span>
      
    </div>
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <!--v-if-->
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      带图标标签
    </p>
    
    <div
      class="tag-demo"
    >
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <svg
            class="t-icon t-icon-app"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M11.25 1.75a3 3 0 100 6 3 3 0 000-6zm-2 3a2 2 0 114 0 2 2 0 01-4 0zM2 3a1 1 0 011-1h3.5a1 1 0 011 1v3.5a1 1 0 01-1 1H3a1 1 0 01-1-1V3zm1 0v3.5h3.5V3H3zM2 9.5a1 1 0 011-1h3.5a1 1 0 011 1V13a1 1 0 01-1 1H3a1 1 0 01-1-1V9.5zm1 0V13h3.5V9.5H3zM8.5 9.5a1 1 0 011-1H13a1 1 0 011 1V13a1 1 0 01-1 1H9.5a1 1 0 01-1-1V9.5zm1 3.5H13V9.5H9.5V13z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </span>
        <span
          class="t-tag__text"
        >
          
          标签
          
        </span>
        <!--v-if-->
      </span>
    </div>
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <!--v-if-->
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      圆角标签
    </p>
    
    <div
      class="tag-demo"
    >
      
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--round t-tag--dark t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          圆角
          
        </span>
        <!--v-if-->
      </span>
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--mark t-tag--dark t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          半圆
          
        </span>
        <!--v-if-->
      </span>
      
    </div>
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <!--v-if-->
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      可关闭标签
    </p>
    
    <div
      class="tag-demo"
    >
      
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium t-is-closable t-tag--closable"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          标签
          
        </span>
        <span
          class="t-tag__icon-close"
        >
          <svg
            class="t-icon t-icon-close"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </span>
      </span>
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium t-is-closable t-tag--closable"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <svg
            class="t-icon t-icon-app"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M11.25 1.75a3 3 0 100 6 3 3 0 000-6zm-2 3a2 2 0 114 0 2 2 0 01-4 0zM2 3a1 1 0 011-1h3.5a1 1 0 011 1v3.5a1 1 0 01-1 1H3a1 1 0 01-1-1V3zm1 0v3.5h3.5V3H3zM2 9.5a1 1 0 011-1h3.5a1 1 0 011 1V13a1 1 0 01-1 1H3a1 1 0 01-1-1V9.5zm1 0V13h3.5V9.5H3zM8.5 9.5a1 1 0 011-1H13a1 1 0 011 1V13a1 1 0 01-1 1H9.5a1 1 0 01-1-1V9.5zm1 3.5H13V9.5H9.5V13z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </span>
        <span
          class="t-tag__text"
        >
          
          标签
          
        </span>
        <span
          class="t-tag__icon-close"
        >
          <svg
            class="t-icon t-icon-close"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </span>
      </span>
      
    </div>
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <!--v-if-->
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      超长省略标签
    </p>
    
    <div
      class="tag-demo"
    >
      <span
        aria-disabled="false"
        class="t-tag t-tag--default t-tag--square t-tag--dark t-tag--medium"
        role="button"
        style="max-width: 130px;"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          听说超长可以省略听说超长
          
        </span>
        <!--v-if-->
      </span>
    </div>
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <h2
      class="tdesign-mobile-demo-block__title"
    >
      02 状态
    </h2>
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      标签状态
    </p>
    
    <div
      class="tag-demo"
    >
      
      <span
        class="t-tag t-tag--checkable t-tag--square t-tag--medium t-is-checked t-tag--checked"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          选中
          
        </span>
        <!--v-if-->
      </span>
      <span
        class="t-tag t-tag--checkable t-tag--square t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          未选中
          
        </span>
        <!--v-if-->
      </span>
      <span
        aria-disabled="true"
        class="t-tag t-tag--checkable t-tag--square t-tag--medium t-is-disabled t-tag--disabled"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          不可选
          
        </span>
        <!--v-if-->
      </span>
      
    </div>
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <h2
      class="tdesign-mobile-demo-block__title"
    >
      03 规格
    </h2>
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      标签规格
    </p>
    
    <div
      class="group padding-bottom d-flex"
    >
      
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--large"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          展示标签30
          
        </span>
        <!--v-if-->
      </span>
      <span
        aria-disabled="false"
        class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          展示标签24
          
        </span>
        <!--v-if-->
      </span>
      
    </div>
    <div
      class="group d-flex"
    >
      <span
        aria-disabled="false"
        class="t-tag t-tag--checkable t-tag--square t-tag--large"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          点击标签30
          
        </span>
        <!--v-if-->
      </span>
      <span
        aria-disabled="false"
        class="t-tag t-tag--checkable t-tag--square t-tag--medium"
        role="button"
      >
        <span
          class="t-tag__icon"
        >
          <!---->
        </span>
        <span
          class="t-tag__text"
        >
          
          点击标签24
          
        </span>
        <!--v-if-->
      </span>
    </div>
    
  </div>
</div>
`;

exports[`Tag > Tag shapeVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <span
    aria-disabled="false"
    class="t-tag t-tag--primary t-tag--round t-tag--dark t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      圆角
      
    </span>
    <!--v-if-->
  </span>
  <span
    aria-disabled="false"
    class="t-tag t-tag--primary t-tag--mark t-tag--dark t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      半圆
      
    </span>
    <!--v-if-->
  </span>
  
</div>
`;

exports[`Tag > Tag sizeVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <span
    aria-disabled="false"
    class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--large"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      展示标签30
      
    </span>
    <!--v-if-->
  </span>
  <span
    aria-disabled="false"
    class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      展示标签24
      
    </span>
    <!--v-if-->
  </span>
  
</div>
`;

exports[`Tag > Tag themeVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <span
    aria-disabled="false"
    class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      标签
      
    </span>
    <!--v-if-->
  </span>
  <span
    aria-disabled="false"
    class="t-tag t-tag--danger t-tag--square t-tag--dark t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      标签
      
    </span>
    <!--v-if-->
  </span>
  <span
    aria-disabled="false"
    class="t-tag t-tag--warning t-tag--square t-tag--dark t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      标签
      
    </span>
    <!--v-if-->
  </span>
  <span
    aria-disabled="false"
    class="t-tag t-tag--success t-tag--square t-tag--dark t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      标签
      
    </span>
    <!--v-if-->
  </span>
  <span
    aria-disabled="false"
    class="t-tag t-tag--default t-tag--square t-tag--dark t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      标签
      
    </span>
    <!--v-if-->
  </span>
  
</div>
`;

exports[`Tag > Tag variantVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <span
    aria-disabled="false"
    class="t-tag t-tag--primary t-tag--square t-tag--dark t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      深色
      
    </span>
    <!--v-if-->
  </span>
  <span
    aria-disabled="false"
    class="t-tag t-tag--primary t-tag--square t-tag--light t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      浅色
      
    </span>
    <!--v-if-->
  </span>
  <span
    aria-disabled="false"
    class="t-tag t-tag--primary t-tag--square t-tag--outline t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      描边
      
    </span>
    <!--v-if-->
  </span>
  <span
    aria-disabled="false"
    class="t-tag t-tag--primary t-tag--square t-tag--light-outline t-tag--medium"
    role="button"
  >
    <span
      class="t-tag__icon"
    >
      <!---->
    </span>
    <span
      class="t-tag__text"
    >
      
      浅色描边
      
    </span>
    <!--v-if-->
  </span>
  
</div>
`;
